File: 프로그래머스/0/181911. 부분 문자열 이어 붙여 문자열 만들기/부분 문자열 이어 붙여 문자열 만들기.java
class Solution {
    public String solution(String[] my_strings, int[][] parts) {
        String answer = "";
        
        for(int i = 0; i < my_strings.length; i++) {
            answer += my_strings[i].substring(parts[i][0], parts[i][1] + 1);
        }
        
        return answer;
    }
}

// 다른 풀이
 return IntStream.range(0, myStrings.length)
                 .mapToObj(i -> myStrings[i].substring(parts[i][0], parts[i][1] + 1))
                 .collect(Collectors.joining());

File: 프로그래머스/0/181911. 부분 문자열 이어 붙여 문자열 만들기/부분 문자열 이어 붙여 문자열 만들기.py
def solution(my_strings, parts):
    answer = ''
    cnt = 0
    
    for string in my_strings:
        answer += string[parts[cnt][0] : parts[cnt][1] + 1]
        cnt += 1
    
    return answer

# 다른 풀이
answer = ""
for i, (s, e) in enumerate(parts):
    answer += my_strings[i][s:e+1]
return answer

# 다른 풀이
return ''.join([x[y[0]:y[1]+1] for x,y in zip(my_strings, parts)])

# 다른 풀이
answer = ''
for s, (x, y) in zip(my_strings, parts):
    answer += s[x:y+1]
return answer
